executor/wasm-host/src/host.rs

```diff
@@ -1485,6 +1485,146 @@ pub fn casper_env_block_time<S: GlobalStateReader, E: Executor>(
     Ok(block_time.value())
 }
 
+pub fn casper_env_info<S: GlobalStateReader, E: Executor>(
+    mut caller: impl Caller<Context = Context<S, E>>,
+    entity_addr_ptr: u32,
+    entity_addr_len: u32,
```

Comment:
There was one comment in the previous PR regarding accepting entity addr arg here. We do have already a function to check balance, so perhaps we should limit the scope of env info to only information available per execution i.e. block height, block hash, etc.

Also, on that note, for future extensibility it'd be helpful to also pass "info_len" value (Wasm will pass `size_of` of the struct), so therefore host will copy only up to "info_size" bytes - in case we'd add more data to this struct this shouldn't break existing wasm stored on chain